affine/chutes.py

```diff
@@ -259,20 +295,40 @@ async def fetch(uid: int):
                     slug = slug,
                     chute=chute,
                 )
+                miner.weights_sha = await get_weights_sha(model, miner_revision)
                 return miner
         except: pass
     results = await asyncio.gather(*(fetch(uid) for uid in uids))
     output = {uid: m for uid, m in zip(uids, results) if m is not None}
     # Remove duplicates.
     if output:
-        best_by_model: Dict[str, Tuple[int, int]] = {}
+        # First, prefer unique by weights_sha: keep earliest block per SHA
```

Comment:
as it's a deep, nested if, early-exit may be better (i.e., `if not output: return output`)